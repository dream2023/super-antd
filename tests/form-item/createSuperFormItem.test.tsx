import React from 'react'
import type { FC } from 'react'
import { render } from '@testing-library/react'
import { createSuperFormItem, createSuperFormItemWithOptions, SuperForm } from 'super-antd'

const BaseDemo: FC<any> = (props) => {
  return <div>{JSON.stringify(props)}</div>
}

const Demo1 = createSuperFormItem(BaseDemo)
const Demo2 = createSuperFormItemWithOptions(BaseDemo)

const App = () => {
  return <SuperForm isResponsive={false}>
    <Demo1 />
    <Demo2 />
  </SuperForm>
}

test('createSuperFormItem', () => {
  const warpper = render(<App />)
  expect(warpper.container).toMatchSnapshot()
})
