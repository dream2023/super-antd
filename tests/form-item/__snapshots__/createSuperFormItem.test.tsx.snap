// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createSuperFormItem 1`] = `
<div>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <div>
        <form
          class="ant-form ant-form-horizontal"
        >
          <input
            style="display: none;"
            type="text"
          />
          <div>
            {"colon":false,"label":" "}
          </div>
          <div>
            {"colon":false,"label":" ","options":[]}
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <div
                    class="ant-space ant-space-horizontal ant-space-align-center super-form-btns"
                    style="justify-content: flex-start;"
                  >
                    <div
                      class="ant-space-item"
                      style="margin-right: 8px;"
                    >
                      <button
                        class="ant-btn ant-btn-primary"
                        type="submit"
                      >
                        <span>
                          提 交
                        </span>
                      </button>
                    </div>
                    <div
                      class="ant-space-item"
                    >
                      <button
                        class="ant-btn"
                        type="reset"
                      >
                        <span>
                          重 置
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
